Source/Analytics/Read/Map/MapProcessor.cs

```diff
@@ -0,0 +1,91 @@
+using Dolittle.Events.Processing;
+using Dolittle.ReadModels;
+using Events.Reporting.CaseReports;
+using Concepts.HealthRisk;
+using System;
+using Concepts;
+using System.Collections.Generic;
+
+namespace Read.Map
+{
+    public class MapProcessor : ICanProcessEvents
+    {
+        readonly IReadModelRepositoryFor<CaseReportsBeforeDay> _caseReportRepository;
+
+        public MapProcessor(IReadModelRepositoryFor<CaseReportsBeforeDay> caseReportRepository){
+            _caseReportRepository = caseReportRepository;
+        }
+
+        [EventProcessor("f52d714c-ca1a-460f-abff-f585d7f98df8")]
+        public void Process(CaseReportReceived @event){
+
+            // Each day contains a health risk dictionary
+            // Each healthrisk in the healthrisk dictionary contains two collections of case-reports from 7 and 30 days 
+            var today = Day.Today;
+            var totalCases = @event.NumberOfFemalesAged5AndOlder + @event.NumberOfFemalesUnder5 + @event.NumberOfMalesAged5AndOlder + @event.NumberOfMalesUnder5;
+            var locationForCaseReport = new Location(@event.Latitude, @event.Longitude);
+            var healthRiskId = @event.HealthRiskId;
+
+            var dayOfCaseReport = Day.From(@event.Timestamp);
+            var timeLimit30Days = dayOfCaseReport + 30;  
+            var timeLimit7Days  = dayOfCaseReport + 7;
+
+            // Insert or update casereport depending on if it is in the list of not             
+            for(Day i = dayOfCaseReport; i < timeLimit30Days; i++){
```

Comment:
I think we can simply this a bit. I propose that we look at it together today :)